app/models/order.rb

```diff
@@ -4,7 +4,35 @@ class Order <ApplicationRecord
   has_many :item_orders
   has_many :items, through: :item_orders
 
+
+  enum status:{pending: 0, packaged: 1, shipped: 2, cancelled: 3}
+
+  def self.order_by_status
+    order(:status)
+  end
+
   def grandtotal
     item_orders.sum('price * quantity')
   end
+
+  def total_count
+    item_orders.sum('quantity')
+  end
+
+  def cancel_order
+    update(status: 3)
+    item_orders.update(status: :Unfulfilled)
+  #Any item quantities in the order that were previously fulfilled have their quantities returned to their respective merchant's inventory for that item.
```

Comment:
I think this is  the one issue that Krista still had.  This will be addressed in US50.

Reply:
Yep, this is the piece that goes with User Stories 49-51.